intel-sgx/dcap-artifact-retrieval/src/provisioning_client/mod.rs

```diff
@@ -455,3 +501,136 @@ mod test_helpers {
         String::from_utf8_lossy(&name.value()).to_string()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::string::String;
+
+
+    struct MockService;
+
+    #[derive(Hash, Clone, PartialEq, Eq, Debug)]
+    struct MockInput(u64);
+
+    impl WithApiVersion for MockInput {
+        fn api_version(&self) -> PcsVersion {
+            PcsVersion::V3
+        }
+    }
+
+    #[derive(Clone, PartialEq, Eq, Debug)]
+    struct MockOutput(String);
+
+    impl<'a> ProvisioningServiceApi<'a> for MockService {
+        type Input = MockInput;
+        type Output = MockOutput;
+
+        fn build_request(&'a self, _input: &Self::Input) -> Result<(String, Vec<(String, String)>), Error> {
+            Ok((_input.0.to_string(), vec![]))
+        }
+
+        fn validate_response(&'a self, _code: StatusCode) -> Result<(), Error> {
+            Ok(())
+        }
+
+        fn parse_response(&'a self, response_body: String, _response_headers: Vec<(std::string::String, std::string::String)>, _api_version: PcsVersion) -> Result<Self::Output, Error> {
+            Ok(MockOutput(response_body))
+        }
+    }
+
+    struct MockFetcher;
+
+    impl<'req> Fetcher<'req> for MockFetcher {
+        type Request = String;
+        type Response = String;
+
+        fn build_request(&'req self, url: &String, _headers: Vec<(String, String)>) -> Result<Self::Request, Error> {
+            Ok(url.clone())
+        }
+
+        fn send(&'req self, request: Self::Request) -> Result<(StatusCode, Self::Response), Error> {
+            Ok((StatusCode::Ok, request))
```

Comment:
Nitpick: If you instead of returning the request as the response, change it slightly (e.g., return `Ok((StatusCode::Ok, "response to: ".to_string() + &request))`) you might even catch other subtle bugs.